src/classrooms/classrooms.service.ts

```diff
@@ -51,20 +53,31 @@ export class ClassroomsService {
   }
 
   async getClassroomsForUser(userId: number): Promise<Classroom[]> {
-    const user = await this.em.findOne(User, { id: userId }, { populate: ["teacher"] });
-    if (!user) {
-      throw new NotFoundException("User not found");
-    }
-
-    const classrooms = await this.em.find(
-      Classroom,
-      { teacher: user.teacher },
-      {
-        orderBy: { classTime: "DESC" },
-        populate: ["teacher"],
-      },
+    const user = await this.userRepository.findOneOrFail(
+      { id: userId },
+      { populate: ["teacher", "student"] },
     );
-    // TODO: query for fetching classrooms that a student is enrolled in.
+    let classrooms: Classroom[] = [];
+    if (user.student) {
+      const classroomIds = await this.enrollmentsRepository.getClassroomIdsForStudent(
+        user.student.id,
+      );
+      classrooms = await this.classroomsRepository.find(
+        { id: { $in: classroomIds } },
+        {
+          populate: ["teacher", "teacher.user"],
+          orderBy: { classTime: "DESC" },
+        },
+      );
```

Comment:
[NIT] Does it make sense to abstract this whole snippet into its own method named something like `getClassroomsOfStudent`?